tech_soln/real_time_dashboards.rst

```diff
@@ -0,0 +1,422 @@
+volume is also high, and both historical and live data are important.
+
+In this technical solution we'll demonstrate how to build part of the first example,
+but this architecture would work equally well for the second and many other use-cases.
+
+Running It Yourself
+-------------------
+
+There will be code snippets in this tutorial but they don't specify a complete system.
+There's `a github repo <https://github.com/citusdata/realtime-dashboards-resources>`_ with
+all the details in one place. If you've followed our installation instructions for running
+Citus on either a single or multiple machines you're ready to try it out.
+
+Data Model
+----------
+
+The data we're dealing with is an immutable stream of log data. We'll insert directly into
+Citus but it's also common for this data to first be routed through something like Kafka.
+Doing so has the usual advantages, and makes it easier to pre-aggregate the data once data
+volumes become unmanageably high.
+
+We'll use a simple schema for ingesting HTTP event data. This schema serves as an example
+to demonstrate the overall architecture; a real system might use additional columns.
+
+.. code-block:: sql
+
+  -- this is run on the master
+  CREATE TABLE http_request (
+    site_id INT,
+    ingest_time TIMESTAMPTZ DEFAULT now(),
+
+    url TEXT,
+    request_country TEXT,
+    ip_address INET,
+
+    status_code INT,
+    response_time_msec INT
+  );
+  SELECT master_create_distributed_table('http_request', 'site_id', 'hash');
+  SELECT master_create_worker_shards('http_request', 16, 2);
+
+When we call :ref:`master_create_distributed_table <master_create_distributed_table>`
+we ask Citus to hash-distribute ``http_request`` using the ``site_id`` column. That means
+all the data for a particular site will live in the same shard.
+
+When we call :ref:`master_create_worker_shards <master_create_worker_shards>` we tell
+Citus to create 16 shards and 2 replicas of each shard (for a total of 32 shard replicas).
+We recommend :ref:`using 2-4x as many shards <faq_choose_shard_count>` as CPU cores in
+your cluster. Using this many shards lets you rebalance data across your cluster after
+adding new worker nodes.
+
+Using a replication factor of 2 means every shard is held on multiple workers. When a
+worker fails the master will prevent downtime by serving queries for that worker's shards
+using the other replicas.
+
+.. NOTE::
+
+  In Citus Cloud you must use a replication factor of 1 (instead of the 2 used here). As
+  Citus Cloud uses `streaming replication
+  <https://www.postgresql.org/docs/current/static/warm-standby.html>`_ to achieve high
+  availability maintaining shard replicas would be redundant.
+
+With this, the system is ready to accept data and serve queries! We've provided `a data
+ingest script
+<https://github.com/citusdata/realtime-dashboards-resources/blob/master/ingest_example_data.sql>`_
```

Comment:
Data ingest script tries to ingest data as if table has session_id column, but it is not mentioned anywhere in the reference architecture document. Either add it to here or remove from data ingest script.